src/main/groovy/com/okta/test/mock/wiremock/TestUtils.groovy

```diff
@@ -39,4 +45,27 @@ class TestUtils {
         System.arraycopy(bigBytes, startSrc, resizedBytes, startDst, len);
         return resizedBytes;
     }
+
+    static Map<String, List<String>> parseQuery(String query) {
+
+        if (query == null || query.empty) {
+            return null
+        }
+
+        Map<String, List<String>> result = new HashMap<>()
+        Arrays.stream(query.split('&'))
+        .map { it.split('=', 2) }
+        .map { it.eachWithIndex { def entry, int i -> it[i] = URLDecoder.decode(entry, StandardCharsets.UTF_8.toString())}}
+        .forEach {
+            List<String> values = result.getOrDefault(it[0], new ArrayList<String>())
+            if(it.length == 2) {
```

Comment:
It might be useful to add a comment here to specify why we're taking the second list-item here.

Also - is it necessary to return `null` here? Would it make more sense to add it to the result if `it.length == 2`?

```java
if (it.length == 2) {
  // We shouldn't need to use `getOrDefault`
  result.put(it[0], it[1])
}
```

Reply:
There are four possible options here (and its crazy that URL parsing isn't baked into the URL class, so this is a bit quick/dirty, but query params could be:
- `/some-url?foo=bar`
- `/some-url?foo`
- `/some-url?foo=bar1&foo=bar2`
- `/some-url`

`getOrDefault` just ensures there is a result returned, and it will be used at least once per key. It's basically converting a map with duplicate keys, to a map were the values is a collection.

I can add a comment to make that more clear though